Improve CCM listing behavior for GNP object

ngcxy opened the pull request:
The original CCM triggers listing for GNP in an unnecessary frequency.
This change modify the listing logic.

liuyuan10 commented:
/lgtm

bowei commented:
How does this fix the frequent listing? If you are fixing something, the Git commit (not PR) should describe how the fix works.

ngcxy commented:
> How does this fix the frequent listing? If you are fixing something, the Git commit (not PR or anything on Github) should describe how the fix works.

Thanks for suggesting. Commit message modified.

liuyuan10 commented:
suggestion on the commit message:

title: Fix unintended API List call in gnp controller 

The GNP controller is using client-go client to list GNP objects, causing unnecessary API server load. Switched to use cached lister

ngcxy commented:
> suggestion on the commit message:
> 
> title: Fix unintended API List call in gnp controller
> 
> The GNP controller is using client-go client to list GNP objects, causing unnecessary API server load. Switched to use cached lister

Updated. Thanks for the suggestion.

bowei commented:
/lgtm
/approve

bowei commented:
/test
